vllm/lora/layers.py

```diff
@@ -63,6 +64,61 @@ def dec(*args, **kwargs):
     return dec
 
 
+def custom_bgmv(
+    y: torch.Tensor,
+    x: torch.Tensor,
+    wa_t_all: torch.Tensor,
+    wb_t_all: torch.Tensor,
+    indices: torch.LongTensor,
+    layer_idx: int,
+    scale: float,
+):
+    """
+    `wa_t_all` and `wb_t_all` contains all LoRA A and LoRA B weight matrices
+    stacked into a single tensor, assuming same rank. The corresponding LoRA
+    A and B for each sample is selected based on `indices`. This avoids a
+    for-loop as well as graph breaks.
```

Comment:
Will be good to add a small example here with wa_t_all with 2 lora adapters and how does wa look like after a gather.

Reply:
Done